ray/tests/conftest.py

```diff
@@ -0,0 +1,169 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import copy
+import os
+import time
+from contextlib import contextmanager
+from urllib.parse import urljoin
+
+import pytest
+import requests
+
+from datadog_checks.dev import EnvVars, TempDir, docker_run
+from datadog_checks.dev._env import get_state, save_state
+from datadog_checks.dev.conditions import CheckEndpoints, WaitFor
+from datadog_checks.dev.http import MockResponse
+from datadog_checks.ray import RayCheck
+
+from .common import (
+    E2E_METADATA,
+    HEAD_DASHBOARD_PORT,
+    HEAD_INSTANCE,
+    HEAD_METRICS_PORT,
+    HEAD_OPENMETRICS_ENDPOINT,
+    HERE,
+    MOCKED_HEAD_INSTANCE,
+    MOCKED_WORKER_INSTANCE,
+    RAY_VERSION,
+    SERVE_PORT,
+    SERVE_URL,
+    WORKER1_INSTANCE,
+    WORKER1_METRICS_PORT,
+    WORKER1_OPENMETRICS_ENDPOINT,
+    WORKER2_INSTANCE,
+    WORKER2_METRICS_PORT,
+    WORKER2_OPENMETRICS_ENDPOINT,
+    WORKER3_INSTANCE,
+    WORKER3_METRICS_PORT,
+    WORKER3_OPENMETRICS_ENDPOINT,
+)
+
+
+@pytest.fixture(scope='session')
+def dd_environment():
+    with docker_run(
+        compose_file=os.path.join(
+            HERE,
+            "docker",
+            "docker-compose.yaml",
+        ),
+        env_vars={
+            "RAY_VERSION": RAY_VERSION,
+            "HEAD_METRICS_PORT": HEAD_METRICS_PORT,
+            "HEAD_DASHBOARD_PORT": HEAD_DASHBOARD_PORT,
+            "WORKER1_METRICS_PORT": WORKER1_METRICS_PORT,
+            "WORKER2_METRICS_PORT": WORKER2_METRICS_PORT,
+            "WORKER3_METRICS_PORT": WORKER3_METRICS_PORT,
+            "SERVE_PORT": SERVE_PORT,
+        },
+        conditions=[
+            CheckEndpoints(HEAD_OPENMETRICS_ENDPOINT),
+            CheckEndpoints(WORKER1_OPENMETRICS_ENDPOINT),
+            CheckEndpoints(WORKER2_OPENMETRICS_ENDPOINT),
+            CheckEndpoints(WORKER3_OPENMETRICS_ENDPOINT),
+            CheckEndpoints(urljoin(SERVE_URL, "hello")),
+            WaitFor(run_add),
+        ],
+        wrappers=[create_log_volumes()],
+    ):
+        # Exercise the service a bit
+        for _ in range(10):
+            run_add()
+            time.sleep(1)
```

Comment:
Let's put this in another condition at the end so stopping an environment doesn't also run this part